File: src/main/java/io/github/shirohoo/library/application/persistance/user/UserLoanHistoryRepositoryImpl.java
package io.github.shirohoo.library.application.persistance.user;

import io.github.shirohoo.library.domain.user.UserLoanHistoryRepository;
import org.springframework.stereotype.Repository;

@Repository
public class UserLoanHistoryRepositoryImpl implements UserLoanHistoryRepository {
    private final UserLoanHistoryJpaRepository jpaRepository;

    public UserLoanHistoryRepositoryImpl(UserLoanHistoryJpaRepository jpaRepository) {
        this.jpaRepository = jpaRepository;
    }

    @Override
    public boolean existsBy(String bookTitle, boolean isReturn) {
        return jpaRepository.existsByBookTitleAndIsReturn(bookTitle, isReturn);
    }
}

File: src/main/kotlin/io/github/shirohoo/library/application/persistence/user/UserLoanHistoryRepositoryImpl.kt
package io.github.shirohoo.library.application.persistence.user

import io.github.shirohoo.library.domain.user.UserLoanHistory
import io.github.shirohoo.library.domain.user.UserLoanHistoryRepository
import io.github.shirohoo.library.domain.user.UserLoanStatus
import org.springframework.stereotype.Repository

@Repository
class UserLoanHistoryRepositoryImpl(
    private val jpaRepository: UserLoanHistoryJpaRepository
) : UserLoanHistoryRepository {
    override fun existsBy(bookTitle: String, status: UserLoanStatus): Boolean {
        return jpaRepository.existsByBookTitleAndStatus(bookTitle, status)
    }

    override fun findAllByStatus(status: UserLoanStatus): List<UserLoanHistory> {
        return jpaRepository.findAllByStatus(status)
    }
}